### PR TITLE
feat: add instance migration confirmation modal

### LINKED ISSUE #18574: I can confirm instance migration before starting the migrate operation
### Description
Users need to confirm instance migration by MIGRATE in the confirmation modal. This should prevent the user from triggering unwanted migrations.

### User Story

```Gherkin
Scenario: As an Operate user
    Given I am in instance migration mode (confirm step)
    When I click on "Confirm"
    Then a modal is shown with an input
    When I click "Confirm" in the modal
    Then the modal stays open and shows a validation error
    When I type "MIGRATE" and click "Confirm" in the modal again
    Then the modal is closed and the migration is triggered
```
### Designs
https://www.figma.com/design/tnqak7ht4Hb4VCGygvLNFU/Version-Migration-HFW?node-id=1970%3A165464&t=EpWVp0Rz3Aw2Hajq-1

### Acceptance Criteria

- [ ] A confirmation modal is opening after clicking "Confirm" in confirm step of instance migration mode
- [ ] The modal contains a text input field

When the text input field **does not** contain "MIGRATE" and the user clicks "Confirm":
- [ ] the modal shows a validation error
- [ ] the modal stays open

When the text input field **contains**  "MIGRATE" and the user clicks "Confirm":
- [ ] the modal closes
- [ ] the migrate operation is triggered

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered and added to the product-hub issue risk assessment
- [ ] Cross-team dependencies have been considered

Optional:
- [ ] Design input has been collected by the assignee
### :point_right: Handover Dev to QA 

- Handy resources: 
     BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc : 
     
- Example projects:

- Versions to validate:
> Docker file : in case that it needed to be tested via docker share the version contain the fixed along with version of other services . 

- Release version ( in which version this fixed/feature will be released)

### PR DESCRIPTION
**Pull request by pedesen**
## Description

- Add instance migration confirmation modal
- Update integration/unit and E2E tests

## Related issues

closes #18574

### CODE REVIEW

File: operate/client/src/App/Processes/MigrationView/MigrationSummaryNotification/index.tsx [hunk 1]
```diff
@@ -7,44 +7,16 @@
...
-            from the process definition:{' '}
-            <strong>
-              {`${selectedSourceProcessName} - version ${selectedSourceProcessVersion}`}
-            </strong>{' '}
-            to the process definition:{' '}
-            <strong>
-              {`${selectedTargetProcess?.name} - version ${selectedTargetVersion}`}
-            </strong>
-          </p>
-
```
Comment (pedesen) [inline operate/client/src/App/Processes/MigrationView/MigrationSummaryNotification/index.tsx:47]: This moved to a separate component `MigrationDetails` which is reused also in the the comfirmation modal.

File: operate/client/src/App/Processes/MigrationView/index.test.tsx [hunk 2]
```diff
@@ -211,65 +206,4 @@ describe('MigrationView', () => {
       expect(await screen.findByText(/processes page/i)).toBeInTheDocument();
     },
   );
-
-  it('should render summary notification in summary step', async () => {
```
Comment (pedesen) [inline operate/client/src/App/Processes/MigrationView/index.test.tsx:215]: This test basically moved to `MigrationDetails/index.test.tsx`

**Review by vitorwtavares — APPROVED** [review]

**Comment by pedesen** [comment]
@gastonpillet01 Please check locally using this docker image: `registry.camunda.cloud/team-operate/camunda-operate:branch-fe-18574-confirm-instance-migration`

**Review by gastonpillet01 — APPROVED** [review]
Instance migration confirmation modal was added and works perfectly.

**Review by vitorwtavares — APPROVED** [review]

**Review by Szik — APPROVED** [review]

**Pull request merged**